### PR TITLE
Minor refactor of test/e2e/common names

### PR DESCRIPTION
**Pull request by wojtek-t**
```release-note
NONE
```

/kind cleanup
/priority important-longterm

### CODE REVIEW

**Comment by johnbelamaric** [comment]
Conflict needs resolving. Question: why do we need this?

**Comment by johnbelamaric** [comment]
Oh, I see the why is in the related PR. That's fine. But please clean up the conflict.

File: test/e2e/common/framework.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+*/
+
+package common
+
+import "k8s.io/kubernetes/test/e2e/framework"
+
+// SIGNetworkDescribe annotates the test with the SIG Network label.
+func SIGNetworkDescribe(text string, body func()) bool {
+	return framework.KubeDescribe("[sig-network] "+text, body)
+}
```
Comment (ehashman) [inline test/e2e/common/framework.go:24]: We will have to add a new one of these functions for every SIG, any reason why we don't do a generic one which can take an argument and output `[sig-{arg}]`?

File: test/e2e/node/runtimeclass.go [hunk 2]
```diff
@@ -35,7 +35,7 @@ import (
 	"github.com/onsi/gomega"
 )
 
-var _ = ginkgo.Describe("[sig-node] RuntimeClass", func() {
+var _ = SIGDescribe("RuntimeClass", func() {
```
Comment (dims) [inline test/e2e/node/runtimeclass.go:38]: SIGNodeDescribe right?

Comment (wojtek-t) [reply test/e2e/node/runtimeclass.go:38]: No - in test/e2e/node SIGDescribe already has [sig-node]

File: test/e2e/common/framework.go (hunk 1 above)
Comment (wojtek-t) [reply test/e2e/common/framework.go:24]: `I would say that the cost of adding 3 boiler-plate lines isn't a big cost.
OTOH - that gives compile time guarantees that if we implement that function correctly, the other also will have correct tags.

[I don't feel very strongly about this though...]

**Comment by wojtek-t** [comment]
@johnbelamaric - PTAL

**Comment by wojtek-t** [comment]
/hold

ref https://github.com/kubernetes/kubernetes/pull/98957#issuecomment-779485998

File: test/e2e/node/runtimeclass.go (hunk 2 above)
Comment (SergeyKanzhelev) [reply test/e2e/node/runtimeclass.go:38]: this is a bit confusing, maybe worth keeping the name to SIGNodeDescribe

File: test/e2e/common/framework.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+// SIGNodeDescribe annotates the test with the SIG Node label.
+func SIGNodeDescribe(text string, body func()) bool {
+	return framework.KubeDescribe("[sig-node] "+text, body)
```
Comment (SergeyKanzhelev) [inline test/e2e/common/framework.go:28]: Would it make sense to add a check here that `text` do not have `[sig-` substring? Also same check maybe useful for `ginkgo.By` wrapper

**Comment by SergeyKanzhelev** [comment]
/retest
/lgtm

I like this approach. The next step would be to wrap all ginkgo methods to allow adding strongly-typed tags.

File: test/e2e/common/configmap.go [hunk 4]
```diff
@@ -31,7 +31,7 @@ import (
 	"github.com/onsi/ginkgo"
 )
 
-var _ = ginkgo.Describe("[sig-node] ConfigMap", func() {
+var _ = SIGNodeDescribe("ConfigMap", func() {
```
Comment (SergeyKanzhelev) [inline test/e2e/common/configmap.go:34]: this change added the `[k8s.io]` tag to the beginning as well, correct? I don't know the history of this tag, is it expected?

Comment (SergeyKanzhelev) [reply test/e2e/common/configmap.go:34]: oh, I found that discussion that the tag will be removed. nvm.

**Comment by wojtek-t** [comment]
> I like this approach. The next step would be to wrap all ginkgo methods to allow adding strongly-typed tags.

Yeah - we should make a step forward after that, but let's proceed with this and do further steps later.

/hold cancel

@johnbelamaric - can you please take another look?

**Comment by spiffxp** [comment]
I would strongly prefer we remove `SIGFooDescribe` in favor of package-local `SIGDescribe` functions instead of `SIGDescribe(sig)` or `SIGFooDescribe` because it better aligns with our review/approve mechanisms

I think @wojtek-t agrees with this, since he held for it https://github.com/kubernetes/kubernetes/pull/98957#issuecomment-779485998

I just want to make sure there are followup PRs to converge toward that pattern

**Comment by spiffxp** [comment]
Trying to get consensus here https://github.com/kubernetes/kubernetes/issues/98326#issuecomment-782744493

**Comment by wojtek-t** [comment]
> I think @wojtek-t agrees with this, since he held for it #98957 (comment)
> I just want to make sure there are followup PRs to converge toward that pattern

Yes - I will ensure follow-ups will happen once we agree it's the way to go (which I personally support).

**Pull request merged**